build-system/tasks/performance/measure-documents.js

```diff
@@ -148,32 +243,43 @@ function writeMetrics(url, version, metrics) {
  * @param {string} url
  * @param {string} version "control" or "experiment"
  * @param {Object} options
+ * @param {Object} handlers
  * @return {Promise}
  */
-async function measureDocument(url, version, {headless}) {
+async function measureDocument(url, version, {headless}, handlers) {
   const browser = await puppeteer.launch({
     headless,
     args: [
       '--allow-file-access-from-files',
       '--enable-blink-features=LayoutInstabilityAPI',
+      '--disable-web-security',
     ],
   });
 
   const page = await browser.newPage();
+  const {handlerOptions, handlerName} = getHandlerFromUrl(url, handlers);
   await page.setCacheEnabled(false);
   await setupMeasurement(page);
 
+  if (handlerOptions) {
+    await page.setRequestInterception(true);
+    setUpRequestHandler(handlerOptions, handlerName, page);
+  }
+
   try {
     await page.goto(`file:${urlToCachePath(url, version)}`, {
       waitUntil: 'networkidle0',
     });
-  } catch {
+  } catch (e) {
     // site did not load
+    console.log(e);
     await browser.close();
     return;
   }
 
-  const metrics = await readMetrics(page);
+  let metrics = await readMetrics(page);
+  await delayBasedOnHandlerOptions(handlerOptions);
```

Comment:
If I read it correctly the process looks like this
wait for all performance metrics to be collected
wait for another timeout value no matter what.

While I'm not sure if there's a timeout for `readMetrics`, but I think we should definitely return early once the handler metric has already been collected without waiting for the timeout.

Reply:
Ok that sounds good to me.